tests/Makefile

```diff
@@ -48,44 +39,35 @@ keyboard_interface: cli/keyboard_interface.c ../net_handler/*.* client/net_handl
 
 ################################ VIRTUAL DEVICES ###############################
 
-devices: general_dev simple_dev unresponsive_dev foreign_dev unstable_dev time_dev sound_dev
+VIRTUAL_DEVICE_FILES := client/virtual_devices/virtual_device_util.* ../dev_handler/message.* ../runtime_util/runtime_util.* ../logger/logger.c
 
-VIRTUAL_DEVICE_FILES := client/virtual_devices/virtual_device_util.o ../dev_handler/message.o ../runtime_util/runtime_util.o ../logger/logger.o
+VIRTUAL_DEVICES := $(shell ls client/virtual_devices/*Device.c | awk -F'.' '{ print $$1 }')
 
-general_dev: client/virtual_devices/GeneralTestDevice
+devices: $(VIRTUAL_DEVICES)
+	echo "virtual devices made" > devices
```

Comment:
`gcc` will always repeat what it does which is why you see that. but this does work as intended as you can only do `make devices` once. This was the hack I mentioned earlier because we create a file called `devices`.